pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -71,12 +71,18 @@ func (s *CloudWatchService) Init() error {
 }
 
 func (s *CloudWatchService) NewExecutor(*models.DataSource) (plugins.DataPlugin, error) {
-	return newExecutor(s.LogsService), nil
+	config := setting.GetCfg()
```

Comment:
I know we want to get rid of global state, so therefore injecting the setting config into the executor like this. Unnecessary?

Reply:
@sunker unless I'm overlooking something, you should use DI to inject `setting.Cfg` into `CloudWatchService`. That's the normal way of doing it, and I think it should work just fine.

Reply:
Okay will do!